src/main/rx/srxl2.c

```diff
@@ -0,0 +1,610 @@
+        setRssi(RSSI_MAX_VALUE / 2, RSSI_SOURCE_RX_PROTOCOL);
+    }
+
+    if (channelData->rssi == 0) {
+        globalResult = RX_FRAME_FAILSAFE;
+    } else {
+        globalResult = RX_FRAME_COMPLETE;
+    }
+
+    const uint16_t *frameChannels = (const uint16_t *) (channelData + 1);
+
+    for(size_t i = 0; i<4; i++)
+    {
+        if (channelData->channelMask.u8[i]) {
+            for (size_t j = 0; j < 8; ++j) {
+                if (channelData->channelMask.u8[i]>> j & 1) {
+                    rxRuntimeConfig->channelData[i*8+j] = *frameChannels++;
+                }
+            }
+        }
+    }
+
+    // DEBUG("channel data: %d %d %x\r\n",
+    //     channelData_header->rssi, channelData_header->frameLosses, channelData_header->channelMask.u32);
+}
+
+bool srxl2ProcessControlData(const Srxl2ControlDataSubHeader* controlData, rxRuntimeConfig_t *rxRuntimeConfig)
+{
+    const uint8_t ownId = (FlightController << 4) | unitId;
+    if (controlData->replyId == ownId) {
+        telemetryRequested = true;
+//        DEBUG("command: %x replyId: %x ownId: %x\r\n", controlData->command, controlData->replyId, ownId);
+    }
+
+    switch (controlData->command) {
+    case ChannelData: {
+        srxl2ProcessChannelData((const Srxl2ChannelDataHeader *) (controlData + 1), rxRuntimeConfig);
+    } break;
+
+    case FailsafeChannelData: {
+        srxl2ProcessChannelData((const Srxl2ChannelDataHeader *) (controlData + 1), rxRuntimeConfig);
+        setRssiDirect(0, RSSI_SOURCE_RX_PROTOCOL);
+        // DEBUG("fs channel data\r\n");
+    } break;
+
+    case VTXData: {
+#if defined(USE_SPEKTRUM_VTX_CONTROL) && defined(USE_VTX_COMMON)
+        //DEBUG("vtx data\r\n");
+        Srxl2VtxData *vtxData = (Srxl2VtxData*)(controlData + 1);
+        //DEBUG("vtx band: %x\r\n", vtxData->band);
+        //DEBUG("vtx channel: %x\r\n", vtxData->channel);
+        //DEBUG("vtx pit: %x\r\n", vtxData->pit);
+        //DEBUG("vtx power: %x\r\n", vtxData->power);
+        //DEBUG("vtx powerDec: %x\r\n", vtxData->powerDec);
+        //DEBUG("vtx region: %x\r\n", vtxData->region);
+        // Pack data as it was used before srxl2 to use existing functions.
+        // Get the VTX control bytes in a frame
+        uint32_t vtxControl =   (0xE0 << 24) | (0xE0 << 8) |
+                                ((vtxData->band & 0x07) << 21) |
+                                ((vtxData->channel & 0x0F) << 16) |
+                                ((vtxData->pit & 0x01) << 4) |
+                                ((vtxData->region & 0x01) << 3) |
+                                ((vtxData->power & 0x07));
+        spektrumHandleVtxControl(vtxControl);
+#endif
+    } break;
+    }
+
+    return true;
+}
+
+bool srxl2ProcessPacket(const Srxl2Header* header, rxRuntimeConfig_t *rxRuntimeConfig)
+{
+    switch (header->packetType) {
+    case Handshake: return srxl2ProcessHandshake(header, (const Srxl2HandshakeSubHeader *) (header + 1));
```

Comment:
Can we have these on separate lines for consistency with the rest of the code please?

Reply:
Done